src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add new book" + book.getTitle(), e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        try (PreparedStatement statement = ConnectionUtil
+                .getConnection().prepareStatement(SQL_FIND_ROW_BY_ID)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book bookFromDb = mapToBook(resultSet);
+                return Optional.of(bookFromDb);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (PreparedStatement statement = ConnectionUtil
+                .getConnection().prepareStatement(SQL_GET_ALL_ROWS)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Book bookFromDb = mapToBook(resultSet);
+                bookList.add(bookFromDb);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (PreparedStatement preparedStatement = ConnectionUtil
+                .getConnection().prepareStatement(SQL_UPDATE_ROW)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            preparedStatement.setLong(3, book.getId());
+
+            int affectedRows = preparedStatement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to update at least 1 row, but was: "
+                        + affectedRows);
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update this book value: "
+                    + book.getTitle(), e);
+        }
+    }
+
+    @Override
+    public Boolean deleteById(Long id) {
+        try (PreparedStatement statement = ConnectionUtil
+                .getConnection().prepareStatement(SQL_DELETE_ROW_BY_ID)) {
+            statement.setLong(1, id);
+            int resultSet = statement.executeUpdate();
+            return resultSet > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book with this id:" + id, e);
+        }
+    }
+
+    private Book mapToBook(ResultSet resultSet) {
+        try {
+            String title = resultSet.getString("title");
+            BigDecimal price = resultSet.getBigDecimal("price");
+            Long id = resultSet.getLong("id");

```

Comment:
it is better to use `getObject()` for id, since `getLong()` will return 0 if id is absent